### PR TITLE
Faster index creation

### PR DESCRIPTION
**Pull request by graetzer**
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/3201/

### CODE REVIEW

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/3158/

File: arangod/RocksDBEngine/RocksDBCollection.cpp [hunk 1]
```diff
@@ -1369,58 +1366,85 @@ arangodb::Result RocksDBCollection::fillIndexes(
...
       if (!s.ok()) {
         res = rocksutils::convertStatus(s, rocksutils::StatusHint::index);
         break;
       }
     }
+    
     batch.Clear();
   }
+  
+//  if (res.ok()) {
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBCollection.cpp]: let's remove this if it is not needed.

File: lib/ApplicationFeatures/RocksDBOptionFeature.cpp [hunk 2]
```diff
@@ -69,7 +69,7 @@ RocksDBOptionFeature::RocksDBOptionFeature(
       _tableBlockSize(std::max(rocksDBTableOptionsDefaults.block_size, static_cast<decltype(rocksDBTableOptionsDefaults.block_size)>(16 * 1024))),
       _recycleLogFileNum(rocksDBDefaults.recycle_log_file_num),
       _compactionReadaheadSize(2 * 1024 * 1024),//rocksDBDefaults.compaction_readahead_size
-      _level0CompactionTrigger(2),
+      _level0CompactionTrigger(rocksDBDefaults.level0_file_num_compaction_trigger),
```
Comment (jsteemann) [inline lib/ApplicationFeatures/RocksDBOptionFeature.cpp]: have we tested yet if this makes any difference?

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2070/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2074/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/2081/

**Pull request merged**